sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -544,23 +514,29 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
    * @throws `RestError` with code that is a value from the standard set of HTTP status codes as documented at
    * https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=netframework-4.8
    */
-  async updateQueue(queueName: string, queueOptions: QueueOptions): Promise<UpdateQueueResponse> {
+  async updateQueue(queue: QueueDescription): Promise<UpdateQueueResponse> {
     log.httpAtomXml(
-      `Performing management operation - updateQueue() for "${queueName}" with options: ${queueOptions}`
+      `Performing management operation - updateQueue() for "${queue.queueName}" with options: ${queue}`
     );
 
-    if (!isJSONLikeObject(queueOptions) || queueOptions === null) {
+    if (!isJSONLikeObject(queue) || queue === null) {
       throw new TypeError(
-        `Parameter "queueOptions" must be an object of type "QueueOptions" and cannot be undefined or null.`
+        `Parameter "queue" must be an object of type "QueueDescription" and cannot be undefined or null.`
       );
     }
 
-    const finalQueueOptions: QueueOptions = {};
-    const getQueueResult = await this.getQueueDetails(queueName);
-    Object.assign(finalQueueOptions, getQueueResult, queueOptions);
+    if (areOptionsUndefined(queue, "queueName")) {
+      throw new TypeError(
+        `Parameter "queue" must be an object of type "QueueDescription" and at least one of the parameters other than queueName must be defined.`
+      );
+    }
+
+    const finalQueueOptions: QueueDescription = { queueName: queue.queueName };
+    const getQueueResult = await this.getQueue(queue.queueName);
```

Comment:
Its odd that we need to get the queue first to update it. @ShivangiReja Does this happen in Track 1 .Net code as well?

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/be83d78940ec87314a648c794e9d72fb89afc75d/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/ManagementClient.cs#L787

.NET is not doing it. I'm removing in JS too.

Reply:
Removed